internal/backend/http_transport.go

```diff
@@ -1,15 +1,25 @@
 package backend
 
 import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
 	"net"
 	"net/http"
 	"time"
 
 	"github.com/restic/restic/internal/debug"
 )
 
-// Transport returns a new http.RoundTripper with default settings applied.
-func Transport() http.RoundTripper {
+// Transport returns a new http.RoundTripper with default settings applied. If
+// a custom rootCertFilename is non-empty, it must point to a valid PEM file,
+// otherwise the function will return an error.
+func Transport(rootCertFilename string) (http.RoundTripper, error) {
+	p, err := certPoolFromFile(rootCertFilename)
```

Comment:
This needs to be guarded with an `if rootCertFilename == ""`.

Reply:
Ooops, you're right, of course. I understand that you to write meant `!=`.